ProjFS.Mac/PrjFSKext/KauthHandler.cpp

```diff
@@ -470,6 +861,10 @@ KEXT_STATIC int HandleVnodeOperation(
             &kauthResult,
             kauthError))
     {
+        if (kauthResult == KAUTH_RESULT_DENY && *kauthError != EBADF)
+        {
+            eventTracer.SetDeniedForCrawler();
```

Comment:
Yes, tracing the reason for decisions is definitely something I'd like to do. As you've noticed, the way the code is currently structured didn't allow that without changes. I guess the suggested change makes this cluster of functions **yet** more complicated (see #900) which is something I really wanted to avoid.

I'm going to take another look at this and see if I can find a slightly cleaner approach.